Cython/Compiler/ExprNodes.py

```diff
@@ -1083,6 +1087,10 @@ def coerce_to(self, dst_type, env):
                 pass
             elif src.constant_result is None:
                 src = NoneNode(src.pos).coerce_to(dst_type, env)
+
+            if dst_type == list_type and src_type == list_type and not dst_type.assignable_from(src_type):
```

Comment:
Why is the list type special enough to merit its own condition here? Or is this still a left-over from your proof-of-concept implementation?